subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -150,136 +150,197 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec {
     }
 
     @Unroll
-    def "re-configuring #change in a nested bean during execution time is detected"() {
+    def "re-configuring #change in nested bean during execution time is detected"() {
+        def fixture = new NestedBeanTestFixture()
+
+        buildFile << fixture.taskWithNestedProperty()
+        buildFile << """
+            task configureTask {
+                doLast {
+                    taskWithNestedProperty.bean = secondBean
+                }
+            }
+            
+            taskWithNestedProperty.dependsOn(configureTask)
+        """
+
+        fixture.prepareInputFiles()
+
+        when:
+        fixture.runTask()
+
+        then:
+        executedAndNotSkipped(fixture.task)
+
+        when:
+        fixture.changeFirstBean(change)
+        fixture.runTask()
+
+        then:
+        skipped(fixture.task)
+
+        when:
+        fixture.changeSecondBean(change)
+        fixture.runTask()
+
+        then:
+        executedAndNotSkipped(fixture.task)
+
+        where:
+        change << ['inputProperty', 'inputFile', 'outputFile']
+    }
+
+    @Unroll
+    def "re-configuring #change in nested bean after the task started executing has no effect"() {
```

Comment:
Does/should it have an effect on outputs?

Reply:
Changing the output file declaration while the task is executing has effect on the task actions if they call the getter for the actual output file property (this is why this test was failing when I used `doFirst` to change the bean).

I think the declaration of the outputs should be considered as fixed as soon as the task starts executing. If it is changed in a task action, then the output file snapshot before execution of the task would be wrong and we would possibly detect the wrong outputs.
WDYT?